build/xtask/src/auxflash.rs

```diff
@@ -0,0 +1,86 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use anyhow::{bail, Context, Result};
+use serde::Deserialize;
+use sha3::{Digest, Sha3_256};
+use zerocopy::AsBytes;
+
+/// List of binary blobs to include in the auxiliary flash binary shipped with
+/// this image.  The auxiliary flash is used to offload storage of large
+/// configuration files (e.g. FPGA bitstreams)
+#[derive(Clone, Debug, Deserialize)]
+pub struct AuxFlash {
+    pub blobs: Vec<AuxFlashBlob>,
+}
+
+/// A single binary blob, encoded into the auxiliary flash file.
+#[derive(Clone, Debug, Deserialize)]
+pub struct AuxFlashBlob {
+    pub file: String,
+    pub compress: bool,
+    pub tag: String,
+}
+
+/// Encode the given data as a tagged TLV-C chunk
+fn data_to_tlvc(tag: &str, data: &[u8]) -> Result<Vec<u8>> {
+    if tag.len() != 4 {
+        bail!("Tag must be a 4-byte value, not '{}'", tag);
+    }
+    let mut out = vec![];
+    let mut header = tlvc::ChunkHeader {
+        tag: tag.as_bytes().try_into().unwrap(),
+        len: 0.into(),
+        header_checksum: 0.into(),
+    };
+    out.extend(header.as_bytes());
+
+    let c = tlvc::compute_body_crc(data);
+
+    out.extend(data);
+    let body_len = out.len() - std::mem::size_of::<tlvc::ChunkHeader>();
+    let body_len = u32::try_from(body_len).unwrap();
+
+    // TLV-C requires the body to be padded to a multiple of four!
+    while out.len() & 0b11 != 0 {
+        out.push(0);
+    }
+    out.extend(c.to_le_bytes());
+
+    // Update the header.
+    header.len.set(body_len);
+    header.header_checksum.set(header.compute_checksum());
+
+    out[..std::mem::size_of::<tlvc::ChunkHeader>()]
+        .copy_from_slice(header.as_bytes());
+    Ok(out)
+}
```

Comment:
This seems like a useful function to add to the tlv-c repo

Reply:
Aha, it looks like I could use `tlvc-text` as a library instead of a command-line tool, so I'll try that in the morning.

Reply:
Fixes are pending https://github.com/oxidecomputer/tlvc/pull/2